ccmbioinfo/lfs
# CARDS-2091: Discard PREMs visits with recent survey events

Add a Clirity Import Processor that discards visits for patients who have been sent a survey in the past 6 months.

Testing instructions:
1. Build and startup using docker per [1780](https://github.com/data-team-uhn/cards/pull/1204) testing instructions
2. In mssql, import visitFilterImportTestSql1.sql
3. Import the visits into sling (http://localhost:8080/Subjects.importClarity). Visits should be created for patient 1 and 2
4. In mssql, import visitFilterImportTestSql2.sql
5. Try importing the visits into sling again. A visit should be created for patient 3. Patient 1's second visit should be discarded so patient 1 should still only have 1 visit.
6. Delete all the patients, import visitFilterImportTestSql1.sql into mssql and then sling again to get back to step 3.
7. Use composum to set the Survey events forms to have an `invitation_sent` answer value for 5 months ago (September 2022)
8. In mssql, import visitFilterImportTestSql2.sql
9. Import the visits into sling
10. Visit should be created for patient 3. Patient 1 should have their second visit discarded for being too recent
11. Delete the patients and re-import visitFilterImportTestSql1.sql into mssql and then sling.
12. Use composum to edit the `invitation_sent` answers to 7 months ago (July 2022)
13. In mssql, import visitFilterImportTestSql2.sql
14. Import the visits into sling again. Patient 1 should get a second visit created as the original visit is now old enough to allow another survey to be sent. Patient 3 should have 1 visit as normal

How to set `invitation_sent`:
- Note down the first few characters of the `jcr:uuid` for the [survey events](http://localhost:8080/bin/browser.html/Questionnaires/Survey%20events) questionnaire and the [invitation_sent](http://localhost:8080/bin/browser.html/Questionnaires/Survey%20events/invitation_sent) question
- Find the forms with `questionnaire` matching the survey events uuid and then find the question with the right `question` for the invitation_sent uuid
- Create a new property (type `Date`, name `value`, value set using the datepicker
